src/drivers/basic/power.c

```diff
@@ -0,0 +1,47 @@
+/**
+ * @section License
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018, Erik Moqvist
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * This file is part of the Simba project.
+ */
+
+#include "simba.h"
+
+#if CONFIG_POWER == 1
+
+#include "power_port.i"
+
+int power_module_init()
+{
+    return (0);
+}
+
+void power_deep_sleep(uint32 time_in_ms)
```

Comment:
Replace `uint32` with `uint32_t`, or possibly `int`, as I usually use `int` for all kind of data. Some people like it, some finds it unusual in embedded programming.

Reply:
Rather uint32_t, as it is in microseconds.

Reply:
This is what it looks like in the thread module: https://github.com/eerimoq/simba/blob/master/src/kernel/thrd.h#L238

Just noticed that the parameter to that function is called `microseconds`, maybe use the same name in the power function to keep it consistent.

Reply:
Ok, so I will change it to `long microseconds`.